apps/cpp_rpc/rpc_env.cc

```diff
@@ -164,7 +164,8 @@ void LinuxShared(const std::string output,
      cmd += " " + *f;
     }
     cmd += " " + options;
-    CHECK(system(cmd.c_str()) == 0) << "Compilation error.";
+    std::string executed_result = common::Execute(cmd);
```

Comment:
You will need to check the status(returned by PClose), instead of the result message, because it could be possible that compilation prints things but still returns 0 status(which means compilation is successful). Please see my original comment